lib/membrane_rtc_engine/endpoints/hls_endpoint.ex

```diff
@@ -290,82 +348,226 @@ if Enum.all?(
           |> to({:opus_decoder, track.id})
           |> to({:aac_encoder, track.id})
           |> to({:aac_parser, track.id})
-          |> via_in(Pad.ref(:input, {:audio, track.id}), options: [encoding: :AAC])
+          |> via_in(Pad.ref(:input, {:audio, track.id}),
+            options: [encoding: :AAC, segment_duration: state.segment_duration]
+          )
           |> to({:hls_sink_bin, track.stream_id})
         ]
       }
     end
 
-    defp hls_links_and_children(link_builder, %Track{encoding: :H264} = track, state) do
-      link_to_transcoder = create_transcoder_link(state.transcoding_config, track.id)
+    if Enum.all?(@audio_mixer_deps, &Code.ensure_loaded?/1) do
+      defp hls_links_and_children(link_builder, %{encoding: :OPUS} = track, state, ctx) do
+        parent_spec = %ParentSpec{
+          children: %{
+            {:track_receiver, track.id} => %TrackReceiver{
+              track: track,
+              initial_target_variant: :high
+            },
+            {:depayloader, track.id} => get_depayloader(track),
+            {:opus_decoder, track.id} => Membrane.Opus.Decoder
+          },
+          links: [
+            link_builder
+            |> to({:track_receiver, track.id})
+            |> to({:depayloader, track.id})
+            |> to({:opus_decoder, track.id})
+            |> to(:audio_mixer)
+          ]
+        }
+
+        state
+        |> generate_audio_mixer(ctx)
+        |> merge_parent_specs(parent_spec)
+      end
+    else
+      defp hls_links_and_children(link_builder, %{encoding: :OPUS}, state, ctx) do
+        raise """
+        Cannot find some of the modules required to use the audio mixer.
+        Ensure that the following dependencies are added to the deps.
+        #{merge_strings(@audio_mixer_deps)}
+        """
+      end
 
+      defp merge_strings(strings),
+        do: strings |> Enum.map(String.replace_suffix("", " ")) |> Enum.reduce(Kernel.<>())
```

Comment:
This function is the same as in check for `compoistor_deps` so I would remove duplicate.

Reply:
What about `Enum.join`?